### ISSUE TITLE
IdentifiersShouldNotContainUnderscores (CA1707) reports on operators

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#947]
#### Analyzer package
Microsoft.ApiDesignGuidelines.Analyzers

#### Analyzer
CA1707: IdentifiersShouldNotContainUnderscores

#### Repro steps

Underscore usage is reported on operators below. Probably because in IL they are named `op_Equality` / `op_Inequality`.

```csharp
public struct S
{
    public static bool operator ==(S left, S right)
    {
        return left.Equals(right);
    }

    public static bool operator !=(S left, S right)
    {
        return !(left == right);
    }
```
#### Expected behavior
No report of CA1707.

#### Actual behavior
CA1707 is reported on both operators.

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/970
